src/Roslyn.Diagnostics.Analyzers/UnitTests/DeclarePublicAPIAnalyzerTests.cs

```diff
@@ -518,5 +538,365 @@ public class C
+        {
+            // Unshipped file has a state 'ObsoleteField' entry and a missing 'NewField' entry.
+            var source = @"
+public class C
+{
+    public int Field;
+    public int Property { get; set; }
+    public void Method() { } 
+    public int ArrowExpressionProperty => 0;
+
+    public int NewField;
+}
+";
+            var shippedText = @"";
+            var unshippedText = @"C
+C.ArrowExpressionProperty.get -> int
+C.C() -> void
+C.Field -> int
+C.Method() -> void
+C.ObsoleteField -> int
+C.Property.get -> int
+C.Property.set -> void";
+            var fixedUnshippedText = @"C
+C.ArrowExpressionProperty.get -> int
+C.C() -> void
+C.Field -> int
+C.Method() -> void
+C.NewField -> int
+C.Property.get -> int
+C.Property.set -> void";
+
+            VerifyCSharpAdditionalFileFix(source, shippedText, unshippedText, fixedUnshippedText);
+        }
+
+        [Fact]
+        public void TestSimpleMissingType_Fix()
+        {
+            var source = @"
+public class C
+{
+    private C() { }
+}
+";
+
+            var shippedText = @"";
+            var unshippedText = @"";
+            var fixedUnshippedText = @"C";
+
+            VerifyCSharpAdditionalFileFix(source, shippedText, unshippedText, fixedUnshippedText);
+        }
+
+        [Fact]
+        public void TestMultipleMissingTypeAndMember_Fix()
+        {
+            var source = @"
+public class C
+{
+    private C() { }
+    public int Field;
+}
+
+public class C2 { }
+";
+
+            var shippedText = @"";
+            var unshippedText = @"";
+            var fixedUnshippedText = @"C
+C.Field -> int
+C2
+C2.C2() -> void";
+
+            VerifyCSharpAdditionalFileFix(source, shippedText, unshippedText, fixedUnshippedText);
+        }
+
+        [Fact]
+        public void TestChangingMethodSignatureForAnUnshippedMethod_Fix()
+        {
+            var source = @"
+public class C
+{
+    private C() { }
+    public void Method(int p1){ }
+}
+";
+
+            var shippedText = @"C";
+            // previously method had no params, so the fix should remove the previous overload.
+            var unshippedText = @"C.Method() -> void";
+            var fixedUnshippedText = @"C.Method(int p1) -> void";
+
+            VerifyCSharpAdditionalFileFix(source, shippedText, unshippedText, fixedUnshippedText);
+        }
+
+        [Fact]
+        public void TestChangingMethodSignatureForAnUnshippedMethodWithShippedOverloads_Fix()
+        {
+            var source = @"
+public class C
+{
+    private C() { }
+    public void Method(int p1){ }
+    public void Method(int p1, int p2){ }
+    public void Method(char p1){ }
+}
+";
+
+            var shippedText = @"C
+C.Method(int p1) -> void
+C.Method(int p1, int p2) -> void";
+            // previously method had no params, so the fix should remove the previous overload.
+            var unshippedText = @"C.Method() -> void";
+            var fixedUnshippedText = @"C.Method(char p1) -> void";
```

Comment:
What would happen if I deleted the unshipped member, and moved it to another class. I guess your analysis would not catch this because it is not a sibling?